File: src/main/java/com/crm/system/models/HistoryMessage.java
package com.crm.system.models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "history_messages")
@Getter
@Setter
public class HistoryMessage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long message_id;

    @NotBlank
    @Column(name = "text_of_message")
    private String messageText;

    @Column(name = "date_of_creation")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dateOfCreation;

    @Column(name = "is_important")
    private boolean isImportant;

    @Column(name = "is_done")
    private boolean isDone;

    private String note;
    @ManyToOne
    @JoinColumn(name = "lid_id")
    @JsonBackReference
    private Client client;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonBackReference
    private User user;

    public HistoryMessage(String messageText) {
        this.messageText = messageText;
        this.dateOfCreation = LocalDateTime.now();
    }
    public HistoryMessage() {
    }
}

File: angular-app/src/entities/HistoryMessage.ts

export interface HistoryMessage {
   messageText: string;
    dateOfCreation: Date;
    isImportant: boolean;
    isDone: boolean;
    note: string;
    lidId: number;
}

const historyWithDateString: HistoryMessage[] = [
    {
      messageText: "Message 1",
      dateOfCreation: new Date("2023-12-10T00:00:00"),
      isImportant: true,
      isDone: false,
      note: "Note 1",
      lidId: 1,
    },
  ];
    